### PR TITLE
[pyright] [core] storage

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Type annotation updates for the largely untyped `dagster._core.storage` package. Also adjusted imports to avoid private module access. There is much more improvement available in this package when/if we upgrade to SQLalchemy 2.0, which is typed.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11339** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11339" target="_blank">Graphite</a> 
    * **PR #11334** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11334" target="_blank">Graphite</a> 
      * **PR #11335** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11335" target="_blank">Graphite</a> 
        * **PR #10900** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10900" target="_blank">Graphite</a> 
          * **PR #10901** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10901" target="_blank">Graphite</a> 
            * **PR #10907** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10907" target="_blank">Graphite</a> 
              * **PR #10992** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10992" target="_blank">Graphite</a> 
                * **PR #10993** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10993" target="_blank">Graphite</a> 
                  * **PR #10994** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10994" target="_blank">Graphite</a> 
                    * **PR #10995** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10995" target="_blank">Graphite</a> 
                      * **PR #10996** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10996" target="_blank">Graphite</a> 
                        * **PR #11329** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11329" target="_blank">Graphite</a> 
                          * **PR #10997** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10997" target="_blank">Graphite</a> 
                            * **PR #10998** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10998" target="_blank">Graphite</a> 
                              * **PR #10999** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10999" target="_blank">Graphite</a> 
                                * **PR #11000** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11000" target="_blank">Graphite</a> 
                                  * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
                                    * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
                                      * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
                                      * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
                                        * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 
                                        * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
                                        * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
                                          * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a>  👈
      * **PR #11212** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11212" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11363?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11339** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11339" target="_blank">Graphite</a> 
    * **PR #11334** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11334" target="_blank">Graphite</a> 
      * **PR #11335** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11335" target="_blank">Graphite</a> 
        * **PR #10900** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10900" target="_blank">Graphite</a> 
          * **PR #10901** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10901" target="_blank">Graphite</a> 
            * **PR #10907** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10907" target="_blank">Graphite</a> 
              * **PR #10992** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10992" target="_blank">Graphite</a> 
                * **PR #10993** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10993" target="_blank">Graphite</a> 
                  * **PR #10994** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10994" target="_blank">Graphite</a> 
                    * **PR #10995** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10995" target="_blank">Graphite</a> 
                      * **PR #10996** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10996" target="_blank">Graphite</a> 
                        * **PR #11329** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11329" target="_blank">Graphite</a> 
                          * **PR #10997** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10997" target="_blank">Graphite</a> 
                            * **PR #10998** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10998" target="_blank">Graphite</a> 
                              * **PR #10999** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10999" target="_blank">Graphite</a> 
                                * **PR #11000** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11000" target="_blank">Graphite</a> 
                                  * **PR #11001** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11001" target="_blank">Graphite</a> 
                                    * **PR #11002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11002" target="_blank">Graphite</a> 
                                      * **PR #11330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11330" target="_blank">Graphite</a> 
                                      * **PR #11038** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11038" target="_blank">Graphite</a> 
                                        * **PR #10983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10983" target="_blank">Graphite</a> 
                                        * **PR #11051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11051" target="_blank">Graphite</a> 
                                        * **PR #11362** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11362" target="_blank">Graphite</a> 
                                          * **PR #11363** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11363" target="_blank">Graphite</a>  👈
      * **PR #11212** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11212" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11363?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/storage/legacy_storage.py [hunk 1]
```diff
@@ -444,7 +479,8 @@ def get_asset_events(
         before_timestamp=None,
         cursor: Optional[int] = None,  # deprecated
     ) -> Union[Iterable["EventLogEntry"], Iterable[Tuple[int, "EventLogEntry"]]]:
-        return self._storage.event_log_storage.get_asset_events(
+        # Type ignore bc method does not exist-- unexecuted code?
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/storage/legacy_storage.py:482]: This approach is excellent. Use the pragma to get it passing but document the knowledge you accrued along that way as a comment 👍🏻

**Review by schrockn — APPROVED** [review]
Big fan of the way you approached the changes here. There were a lot of them but all at the same risk level so easy to review. 👍🏻

File: python_modules/dagster/dagster/_core/storage/legacy_storage.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_core/storage/legacy_storage.py:482]: This is the approach I used where the fix wasn't obvious

**Pull request merged**